### PR TITLE
[Accessibility] Add scope attribute to the product attributes table

### LINKED ISSUE #43608: [Enhancement]: Accessibility: Add scope to products attributes table headers
### Describe the solution you'd like

The product attributes table on the single product is coded with row headers but does not have the scope added to the th tag:
https://github.com/woocommerce/woocommerce/blob/758df4854d7b627fc26eb5c0860dc492dea6a06e/plugins/woocommerce/templates/single-product/product-attributes.php#L29

In VoiceOver, the heading is not clearly announced in the table, particularly if there is only a single row. Adding scope="row" to the TH tags to connect it more explicitly to the TD cell would likely resolve this.

### Describe alternatives you've considered

N/A

### Additional context

_No response_

### LINKED ISSUE #43628: Accessibility: Ambiguous view link on My Account orders table
### Prerequisites

- [X] I have carried out troubleshooting steps and I believe I have found a bug.
- [X] I have searched for similar bugs in both open and closed issues and cannot find a duplicate.

### Describe the bug

On the My Account Orders page, the orders table contains a "View" button in the actions table that does not include any reference to the account number, either in visible text or an aria-label.
https://github.com/woocommerce/woocommerce/blob/758df4854d7b627fc26eb5c0860dc492dea6a06e/plugins/woocommerce/templates/myaccount/orders.php#L70

### Expected behavior

The View link should be unique so that when multiple orders are present in the table, it is possible to distinguish each View link from one another.

Add an aria-label that includes the order number to the link like this:
`<a href="https://shop.equalizedigital.com/my-account/view-order/518/" class="woocommerce-button wp-element-button button view" aria-label="View order number 518">View</a>`

### Actual behavior

Links are ambiguous

### Steps to reproduce

See code linked above.

### WordPress Environment

N/A

### Isolating the problem

- [X] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [X] This bug happens with a default WordPress theme active, or [Storefront](https://woo.com/products/storefront/).
- [X] I can reproduce this bug consistently using the steps above.

### PR DESCRIPTION
**Pull request by amesplant**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

Closes #43608 .

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Visit a Product Detail page, and select the Additional Information Tab
2. Inspect the table to make sure that the row headings have a scope of "row"
3. In addition, a label was added to the table, inspect to make sure that the `aria-label` is "Product Details"
4. Using a screenreader, navigate to the table (Mac VoiceOVer .. use the Rotor to easily and quickly navigate to the table)
5. Navigate through the table and make sure that the first time landing in a table detail, the header is also read off. 
_If the table only has one row, the table detail may not be read as the user will have just heard that single row's heading announced as they navigate through the table._

### Changelog entry

-   [ ] Automatically create a changelog entry from the details below.

-   [ ] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Details</summary>

#### Significance

-   [x] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [x] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>

### CODE REVIEW

File: pnpm-lock.yaml
Comment (dkotter) [inline pnpm-lock.yaml:1]: Were the changes to this file on purpose? Seems like we may want to revert the changes here

File: plugins/woocommerce/templates/single-product/product-attributes.php [hunk 1]
```diff
@@ -23,10 +23,10 @@
 	return;
 }
 ?>
-<table class="woocommerce-product-attributes shop_attributes">
+<table class="woocommerce-product-attributes shop_attributes" aria-label="<?php echo esc_html_e('Product Details', 'woocommerece'); ?>">
```
Comment (dkotter) [inline plugins/woocommerce/templates/single-product/product-attributes.php]: Typo and spacing fixes:

```suggestion
<table class="woocommerce-product-attributes shop_attributes" aria-label="<?php echo esc_html_e( 'Product Details', 'woocommerce' ); ?>">
```

Comment (dkotter) [inline plugins/woocommerce/templates/single-product/product-attributes.php]: The `esc_html_e` function directly echoes things, so no need to have an `echo` statement before that. In addition, wondering if we want to use the `esc_attr` functions here instead since this is an attribute:

```suggestion
<table class="woocommerce-product-attributes shop_attributes" aria-label="<?php esc_attr_e( 'Product Details', 'woocommerce' ); ?>">
```

**Review by dkotter — CHANGES_REQUESTED** [review]

File: plugins/woocommerce/templates/single-product/product-attributes.php (hunk 1 above)
Comment (amesplant) [reply plugins/woocommerce/templates/single-product/product-attributes.php]: Thanks @dkotter good point .. commit accepted ;)

File: pnpm-lock.yaml
Comment (amesplant) [reply pnpm-lock.yaml:1]: @dkotter changes have been reverted

File: plugins/woocommerce/templates/single-product/product-attributes.php [hunk 2]
```diff
@@ -23,10 +23,10 @@
 	return;
 }
 ?>
-<table class="woocommerce-product-attributes shop_attributes">
+<table class="woocommerce-product-attributes shop_attributes" aria-label="<?php echo esc_html_e( 'Product Details', 'woocommerce' ); ?>">
```
Comment (dkotter) [inline plugins/woocommerce/templates/single-product/product-attributes.php]: Think this got missed (which is my fault for adding two separate comments on the same line):

```suggestion
<table class="woocommerce-product-attributes shop_attributes" aria-label="<?php esc_attr_e( 'Product Details', 'woocommerce' ); ?>">
```

**Review by dkotter — CHANGES_REQUESTED** [review]

**Review by dkotter — APPROVED** [review]
@ankitguptaindia @qasumitbagthariya This is ready for QA

**Comment by qasumitbagthariya** [comment]
### QA Update ✅ 
-------------

I have checked this PR in the `amesplant:fix/43608-products-table-headers-scope` branch and the changes are working as expected.


**Testing Environment** 

<Details>

- WordPress: 6.6
- Theme: Storefront 4.6.0
- WooCommerce - WooCommerce - Dev version with this fix specific branch.
- PHP: 8.0.30
- Web Server: Nginx 1.20.2
- Browser: Chrome 
- OS: macOS Ventura 13.3 
- Git Branch: amesplant:fix/43608-products-table-headers-scope

</Details>


https://github.com/user-attachments/assets/8d271310-da00-4952-bf10-965fdd73c6b5


image

image

image

**Comment by qasumitbagthariya** [comment]
@amesplant Please remove [WIP] from the title and take this PR out of the draft.

**Comment by amesplant** [comment]
@qasumitbagthariya removed [Draft] and marked ready for review.

**Review by nielslange — COMMENTED** [review]
Thanks for working on this, @amesplant. While the changes look good, it's better if @WunderBart is reviewing this PR, as he's part of the team that maintains the product page.

File: plugins/woocommerce/changelog/fix-43608-products-table-headers-scope [hunk 3]
```diff
@@ -0,0 +1,4 @@
+Significance: patch
+Type: fix
+
+Add scope attribute to the product attributes table
```
Comment (naman03malhotra) [inline plugins/woocommerce/changelog/fix-43608-products-table-headers-scope]: maybe also add `aria-label` you added as part of this PR.

**Review by naman03malhotra — APPROVED** [review]
Looks good to me! thanks for working on it.

File: plugins/woocommerce/changelog/fix-43608-products-table-headers-scope (hunk 3 above)
Comment (vikrampm1) [reply plugins/woocommerce/changelog/fix-43608-products-table-headers-scope]: @amesplant please add the above to the changelog.

**Comment by vikrampm1** [comment]
@WunderBart this PR is awaiting your review.

File: plugins/woocommerce/changelog/fix-43608-products-table-headers-scope (hunk 3 above)
Comment (amesplant) [reply plugins/woocommerce/changelog/fix-43608-products-table-headers-scope]: @naman03malhotra cc @vikrampm1 updated changelog

**Comment by jeffpaul** [comment]
@naman03malhotra is there anything else needed on this PR before its ready for merge / prepare for a Woo release?

**Review by naman03malhotra — COMMENTED** [review]
Looks good to me, if all the approvals are acquired we can go ahead and merge this!

**Comment by vikrampm1** [comment]
@naman03malhotra this PR has completed all the steps from our end and is ready to be merged by you / Woo unless @WunderBart has any feedback or planning to review as noted by Niels [here](https://github.com/woocommerce/woocommerce/pull/49768#pullrequestreview-2201827038).

**Review by kmanijak — APPROVED** [review]
Tested with the VoiceOver rotor and all seems good, confirming from Woo side it can be merged!
